### PR TITLE
vim-patch:8.2.{56,61,97,98}

### PR DESCRIPTION
**Pull request by VVKot**
- vim-patch:8.2.0056: execution stack is incomplete and inefficient
- vim-patch:8.2.0061: the execute stack can grow big and never shrinks
- vim-patch:8.2.0097: crash with autocommand and spellfile
- vim-patch:8.2.0098: exe stack length can be wrong without being detected

8.2.0056:

`check_map_keycodes` was removed in 3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9.

It seems that `gw_grow` never fails in Neovim - so I made the same assumption when porting `scriptfile.c` code.

`term.c` changes are N/A.

Changes to `exception_defs` are N/A because it was converted to a Lua test in https://github.com/neovim/neovim/pull/8635

I had to extract `exception_defs.h` from `ex_eval.h` because otherwise I was getting an include loop as below. I decided to break it up at the first hop but let me know if you disagree.

typval.h -> ex_eval.h -> ex_cmds_defs.h -> normal.h -> buffer_defs.h -> option_defs.h -> typval.h

8.2.0061:

xrealloc never returns NULL in Neovim so I removed check for that.

8.2.0097

Changed int to boolean.

8.2.0098

No changes in `map.c` as `check_map_keycodes` was removed - see above.

### CODE REVIEW

**Comment by VVKot** [comment]
@janlazo I wasn't sure about porting https://github.com/vim/vim/commit/25e0f5863e9010a75a1ff0d04e8f886403968755 - should I grab all changes besides Vim9 and mark it as partial?

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -8115,16 +8115,16 @@ static void f_rpcrequest(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     // If this is called from a provider function, restore the scope
     // information of the caller.
     save_current_sctx = current_sctx;
-    save_sourcing_name = sourcing_name;
-    save_sourcing_lnum = sourcing_lnum;
+    save_sourcing_name = SOURCING_NAME;
+    save_sourcing_lnum = SOURCING_LNUM;
```
Comment (seandewar) [inline src/nvim/eval/funcs.c:8119]: Shouldn't we be pushing to the execution stack in places like this now?

Comment (VVKot) [reply src/nvim/eval/funcs.c:8119]: Yep, good call - this function is Neovim-specific but we should adapt it as well.

Comment (VVKot) [reply src/nvim/eval/funcs.c]: @seandewar does that (f1dd6a9) look good?

**Comment by VVKot** [comment]
Current errors (`make oldtest TEST_FILE=test_maparg.vim`), `sid` is different. Looking as to why this might be the case - Vim test code looks exact same?

```
Found errors in Test_maparg():
/Users/kot/git-repos/neovim/src/nvim/testdir/runtest.vim line 10: Expected {'lnum': 16, 'script': 0, 'ex
pr': 0, 'noremap': 0, 'lhs': 'foo<C-V>', 'mode': ' ', 'nowait': 0, 'silent': 0, 'sid': 0, 'rhs': 'is<F4>
foo', 'buffer': 0} but got {'lnum': 16, 'script': 0, 'silent': 0, 'noremap': 0, 'lhs': 'foo<C-V>', 'mode
': ' ', 'nowait': 0, 'expr': 0, 'sid': 3, 'rhs': 'is<F4>foo', 'buffer': 0}
/Users/kot/git-repos/neovim/src/nvim/testdir/runtest.vim line 14: Expected {'lnum': 17, 'script': 1, 'ex
pr': 1, 'noremap': 1, 'lhs': 'bar', 'mode': 'v', 'nowait': 0, 'silent': 1, 'sid': 0, 'rhs': 'isbar', 'bu
ffer': 1} but got {'lnum': 17, 'script': 1, 'silent': 1, 'noremap': 1, 'lhs': 'bar', 'mode': 'v', 'nowai
t': 0, 'expr': 1, 'sid': 3, 'rhs': 'isbar', 'buffer': 1}
/Users/kot/git-repos/neovim/src/nvim/testdir/runtest.vim line 20: Expected {'lnum': 28, 'script': 0, 'ex
pr': 0, 'noremap': 0, 'lhs': 'foo', 'mode': ' ', 'nowait': 1, 'silent': 0, 'sid': 0, 'rhs': 'bar', 'buff
er': 1} but got {'lnum': 28, 'script': 0, 'silent': 0, 'noremap': 0, 'lhs': 'foo', 'mode': ' ', 'nowait'
: 1, 'expr': 0, 'sid': 3, 'rhs': 'bar', 'buffer': 1}
/Users/kot/git-repos/neovim/src/nvim/testdir/runtest.vim line 26: Expected {'lnum': 34, 'script': 0, 'ex
pr': 0, 'noremap': 0, 'lhs': 'baz', 'mode': 't', 'nowait': 0, 'silent': 0, 'sid': 0, 'rhs': 'foo', 'buff
er': 0} but got {'lnum': 34, 'script': 0, 'silent': 0, 'noremap': 0, 'lhs': 'baz', 'mode': 't', 'nowait'
: 0, 'expr': 0, 'sid': 3, 'rhs': 'foo', 'buffer': 0}
```

**Comment by justinmk** [comment]
Looks like only `8.2.0098` in HEAD.

**Pull request closed without merging**